client/state/data-layer/wpcom/read/sites/notification-subscriptions/new/test/index.js

```diff
@@ -0,0 +1,75 @@
+/** @format */
+
+/**
+ * Internal dependencies
+ */
+import {
+	requestNotificationSubscription,
+	receiveNotificationSubscriptionError,
+	fromApi,
+} from '../';
+import { bypassDataLayer } from 'state/data-layer/utils';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import {
+	subscribeToNewPostNotifications,
+	unsubscribeToNewPostNotifications,
+} from 'state/reader/follows/actions';
+import { NOTICE_CREATE } from 'state/action-types';
+
+describe( 'notification-subscriptions-new', () => {
+	describe( 'requestNotificationSubscription', () => {
+		test( 'should dispatch an http request', () => {
+			const blogId = 123;
+			const action = subscribeToNewPostNotifications( blogId );
+
+			expect( requestNotificationSubscription( action ) ).toEqual(
+				http(
+					{
+						apiNamespace: 'wpcom/v2',
+						method: 'POST',
+						path: `/read/sites/${ blogId }/notification-subscriptions/new`,
+					},
+					action
+				)
+			);
+		} );
+	} );
+
+	describe( 'fromApi', () => {
+		test( 'should throw an error when success is false', () => {
+			const response = {
+				success: false,
+			};
+
+			expect( () => {
+				fromApi( response );
+			} ).toThrow();
+		} );
+
+		test( 'should throw an error when success is true', () => {
+			const response = {
+				success: true,
+			};
+
+			expect( () => {
```

Comment:
lie!

also, it would probably be better to at least have a check for _what it should do_ as opposed to what it shouldn't.  In this case it would be along the lines of:

`test( 'if response indicates a success, then return response unchanged' )`